recipes/legit/meta.yaml

```diff
@@ -0,0 +1,44 @@
+{% set packagename = 'legit' %}
+{% set version = "0.2.0" %}
+{% set md5 = "5df3cd402b0de8f0251a87904b5eb88c" %}
+
+package:
+  name: {{ packagename }}
+  version: {{ version }}
+
+source:
+  git_url: https://github.com/kennethreitz/legit
+  git_tag: v{{ version }}
+
+build:
+  number: 0
+  script: python setup.py install --single-version-externally-managed --record record.txt
+  entry_points:
+    - legit = legit.cli:main
+
+requirements:
+  build:
+    - python
+    - setuptools
+  run:
+    - python
+    - clint >=0.4.1
+    - gitpython >=0.3.4
+    - six >=1.9.0
+
+test:
+  # Some package might need a `test/commands` key to check CLI.
+  imports:
+    - {{ packagename }}
+
+  commands:
+    - cd {{ environ.SRC_DIR }} && legit help
```

Comment:
OK, now I think I get it.  This is tricky.  I believe it is caused by your cd into the source folder.  In doing so, the legit entry point finds the local legit folder there, rather than the installed one.  Things spiral from there.  I have no good answer for why this does not disrupt the other platforms.  I have confounded myself in this situation before several times, locally, though, and this seems familiar.

In light of this, I still recommend a separate git repo here for testing - cloned or created via gitpython, or perhaps use the conda git package as a test requirement if you need to.  Avoid a cd into the source folder.